vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -95,6 +111,107 @@ def __preprocess(
 
         return request
 
+    def _make_tools_request(
+        self, request: ChatDetectionRequest
+    ) -> Union[ChatDetectionRequest, ErrorResponse]:
+        """Convert original chat detection request to Granite Guardian-compatible request with
+        tools, user, and assistant messages."""
+
+        if (
+            "risk_name" not in request.detector_params
+            or request.detector_params["risk_name"] not in self.TOOLS_RISKS
+        ):
+            # Provide error here, since otherwise follow-on tools message
+            # and assistant message flattening will not be applicable
+            return ErrorResponse(
+                message="tools analysis is not supported with given risk",
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+
+        # 'Flatten' the assistant message, extracting the tool_calls functions
+        # Also check for user messages
+        assistant_message_idxs = []
```

Comment:
looks like the `assistant_message_idxs` are not getting used anywhere except to account for `len`.. Can we instead just have a `count` variable instead of a list then ?